test/unit/tag_test.rb

```diff
@@ -142,7 +142,7 @@ class TagTest < ActiveSupport::TestCase
   end
 
   test 'Tag.top_nodes returns most viewed nodes with specified tag and node type' do
-    top_nodes = Tag.find_top_nodes_by_type(tagname:'awesome2', type:'page')
+    top_nodes = Tag.find_top_nodes_by_type('awesome2', 'page')
```

Comment:
why did you have to change this?

Reply:
Because the format of passing the arguments was wrong. The find_top_nodes_by_type definition had positional arguments and in the tests the arguments were being passed as hash arguments

Reply:
So, I changed the way of passing arguments in the tests

Reply:
can you use please use keyword arguments instead in the implementation of `find_top_nodes_by_type` so we avoid touching the tests? this change is confusing, the `find_top_nodes_by_type` take three args but here we're passing only two of them. keyword arguments should help and make things clearer.

Reply:
ok I will change the arguments in `find_top_nodes_by_type` of `tag.rb` instead